apscheduler/executors/asyncio.py

```diff
@@ -31,29 +43,54 @@ def callback(f):
             else:
                 self._run_job_success(job.id, events)
 
-        events = self._job_runtime(job, run_times)
-        future_events = []
-        for event in events:
-            if not (isinstance(event, asyncio.Future) or asyncio.iscoroutine(event)):
-                future = asyncio.Future()
-                future.set_result(event)
-                future_events.append(future)
-            else:
-                future_events.append(event)
-        future = asyncio.gather(*events)
+        if not asyncio.iscoroutinefunction(job.func):
+            future = self._eventloop.run_in_executor(None, job_runtime, job, run_times, self._logger.name)
+        else:
+            events = job_runtime(job, run_times, self._logger.name, self._run_job)
+            future_events = []
+            for event in events:
+                if not (isinstance(event, asyncio.Future) or asyncio.iscoroutine(event)):
+                    future = asyncio.Future()
+                    future.set_result(event)
+                    future_events.append(future)
+                else:
+                    future_events.append(event)
+            future = asyncio.gather(*events)
         future.add_done_callback(callback)
 
-    @asyncio.coroutine
-    def _run_job(self, job, run_time):
-        """Actual implementation of calling the job function"""
-        try:
-            retval = yield from job.func(*job.args, **job.kwargs)
-        except:
-            exc, tb = sys.exc_info()[1:]
-            formatted_tb = ''.join(format_tb(tb))
-            self._logger.exception('Job "%s" raised an exception', job)
-            return JobExecutionEvent(EVENT_JOB_ERROR, job.id, job._jobstore_alias, run_time,
-                                     exception=exc, traceback=formatted_tb)
-        else:
-            self._logger.info('Job "%s" executed successfully', job)
-            return JobExecutionEvent(EVENT_JOB_EXECUTED, job.id, job._jobstore_alias, run_time, retval=retval)
+    if real_asyncio:
```

Comment:
Native coroutines have the `__await__` method which returns an iterable. On Python < 3.4, you can just iterate over the generator directly:
```python
for v in job.func(*job.args, **job.kwargs):
    retval = yield v
```
I haven't actually done this kind of emulation before but this should help you figure out a common code path that works for both asyncio and trollius plus all versions of Python.